client/packet/reporter.go

```diff
@@ -0,0 +1,146 @@
+package packet
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"runtime"
+
+	"github.com/packethost/pkg/env"
+	"github.com/packethost/pkg/log"
+	"github.com/pkg/errors"
+	"github.com/tinkerbell/boots/client"
+	"github.com/tinkerbell/boots/httplog"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+var _ client.Reporter = &PacketReporter{}
+
+// client has all the fields corresponding to connection
+type PacketReporter struct {
```

Comment:
Might not need to include `Packet` in names in this file with the package being called `packet`.
prefer `packet.Reporter` over `packet.PacketReporter`

"All references to names in your package will be done using the package name, so you can omit that name from the identifiers." --[ref](https://github.com/golang/go/wiki/CodeReviewComments#package-names)